examples/util/WebNNRunner.js

```diff
@@ -0,0 +1,427 @@
+  };
+
+  /**
+   * This method is to do inference with input src (HTML [<img> | <video> | <audio>] element).
+   * @param src: An object for HTML [<img> | <video> | <audio>] element.
+   * @param options: A string to get inputTensor. Details:
+   * options = {
+   *   // inputSize was configed in modelZoo.js, inputSize = [h, w, c] or [1, size] for audio example.
+   *   inputSize: inputSize,
+   *   // preOptions was also configed in modelZoo.js,
+   *   // preOptions= {} or {mean: [number, number, number, number],std: [number, number, number, number]}
+   *   preOptions: preOptions,
+   *   imageChannels: 4,
+   *   drawOptions: { // optional, drawOptions is used for CanvasRenderingContext2D.drawImage() method.
+   *     sx: sx, // the x-axis coordinate of the top left corner of sub-retangle of the source image
+   *     sy: sy, // the y-axis coordinate of the top left corner of sub-retangle of the source image
+   *     sWidth: sWidth, // the width of the sub-retangle of the source image
+   *     sHeight: sHeight, // the height of the sub-retangle of the source image
+   *     dWidth: dWidth, // the width to draw the image in the detination canvas
+   *     dHeight: dWidth, // the height to draw the image in the detination canvas
+   *   },
+   *   scaledFlag: true, // optional, need scaled the width and height of element to get need inputTensor
+   * };
+   * @returns {string} This returns a string for inference status.
+   */
+  run = async (src, options) => {
+    let status = 'ERROR';
+
+    if (src.tagName === 'AUDIO') {
+      await getTensorArrayByAudio(src, this._inputTensor, options);
+    } else {
+      getTensorArray(src, this._inputTensor, options);
+    }
+
+    const start = performance.now();
+    status = await this._model.compute(this._inputTensor, this._outputTensor);
+    const delta = performance.now() - start;
+    this._setInferenceTime(delta);
+    console.log(`Computed Status: [${status}]`);
+    console.log(`Compute Time: [${delta} ms]`);
+    return status;
+  };
+
+  /**
+   * This method is get required ops of model.
+   * @returns {object} This returns an array object for required ops of model.
+   */
+  getRequiredOps = () => {
+    return this._modelRequiredOps;
+  };
+
+  /**
+   * This method is to get inference subgraphs summary info.
+   * @returns {object} This returns an array object for inference subgraphs summary info.
+   */
+  getSubgraphsSummary = () => {
+    return this._subgraphsSummary;
+  };
+
+  /**
+   * This method is to get deQuantize params of deQuantized model.
+   * @returns {object} This returns an array object for deQuantize params of deQuantized model.
+   */
+  getDeQuantizeParams = () => {
+    return this._deQuantizeParams;
+  };
+
+  /**
+   * This method is to output inference tensor for post processing by example side.
+   * @param output: An object for output which will be updated with output tensor info by this method.
+   */
+  _passOutputTensor = (output) => {
```

Comment:
The name of this method looks unclear to me. What does "pass" mean? If it always sets `output.outputTensor`, why not change this method to `_getOutputTensor`, then caller sets `output.outputTensor = _getOutputTensor()`?